Sources/SwiftttCamera.swift

```diff
@@ -142,39 +142,67 @@ extension SwiftttCamera {
+            session.beginConfiguration()  // Begin configuration to setup session
+            session.sessionPreset = .photo  // Set the session preset
+
+            // Use a discovery session to find and prioritize the triple camera
+            let discoverySession = AVCaptureDevice.DiscoverySession(
+                deviceTypes: [
+                    .builtInTripleCamera,
+                    .builtInDualWideCamera,
+                    .builtInDualCamera,
+                    .builtInUltraWideCamera,
+                    .builtInWideAngleCamera,
+                    .builtInTelephotoCamera
+                ],
+                mediaType: .video,
+                position: .back
+            )
+
+            guard let device = discoverySession.devices.first else {
+                print("No compatible device found")
+                session.commitConfiguration()
+                return
+            }
+
             do {
-                try device?.lockForConfiguration()
-                device?.setFocusModeIfSupported(.continuousAutoFocus)
-                device?.setExposureModeIfSupported(.continuousAutoExposure)
-                device?.unlockForConfiguration()
-                #if !targetEnvironment(simulator)
-                let deviceInput: AVCaptureDeviceInput = try  AVCaptureDeviceInput(device: device!)
-                session.addInputIfPossible(deviceInput)
-                switch device!.position {
-                case .back: cameraDevice = .rear
-                case .front: cameraDevice = .front
-                default: break
+                let deviceInput = try AVCaptureDeviceInput(device: device)
+
+                if session.canAddInput(deviceInput) {
+                    session.addInput(deviceInput)
                 }
-                #endif
+
+                if device.isFocusModeSupported(.continuousAutoFocus) {
+                    try device.lockForConfiguration()
+                    device.focusMode = .continuousAutoFocus
+                    device.unlockForConfiguration()
+                }
+
                 photoOutput = AVCapturePhotoOutput()
-                session.addOutputIfPossible(photoOutput)
+                if session.canAddOutput(photoOutput) {
+                    session.addOutput(photoOutput)
+                }
+
+                session.commitConfiguration()  // Commit the session configuration
+
                 deviceOrientation = DeviceOrientation()
+
                 if isViewLoaded && view.window != nil {
                     startRunning()
                     insertPreviewLayer()
                     setPreviewVideoOrientation()
                     resetZoom()
                 }
             } catch {
-                dump(error)
+                print("Error configuring camera device: \(error)")
+                session.commitConfiguration()
             }
         } else {
             delegate?.userDeniedCameraPermissions(forCameraController: self)
         }
     }
 
+
```

Comment:
I prefer a more vertically compact code style 🙏 I'll apply these changes if you don't mind :)

Reply:
FTFY